src/Peering/Peering/PeeringService/GetAzurePeeringServiceCommand.cs

```diff
@@ -0,0 +1,164 @@
+        public string Name { get; set; }
+
+        /// <summary>
+        /// The resource  id
+        /// </summary>
+        [Parameter(
+            Position = 0,
+            Mandatory = true,
+            HelpMessage = Constants.ResourceIdHelp,
+            ParameterSetName = Constants.ParameterSetNameByResourceId)]
+        [ValidateNotNullOrEmpty]
+        public string ResourceId { get; set; }
+
+        /// <inheritdoc />
+        /// <summary>
+        ///     Execute Override for powershell cmdlet
+        /// </summary>
+        public override void Execute()
+        {
+            base.Execute();
+            try
+            {
+                if (string.Equals(
+                    this.ParameterSetName,
+                    Constants.ParameterSetNameByResourceAndName,
+                    StringComparison.OrdinalIgnoreCase))
+                {
+                    if (this.Name != null)
+                    {
+                        var item = this.GetPeeringByResourceAndName();
+                        this.WriteObject(item);
+                    }
+                    else
+                    {
+                        var list = this.GetPeeringByResource();
+                        this.WriteObject(list, true);
+                    }
+                }
+                else if (string.Equals(
+                  this.ParameterSetName,
+                  Constants.ParameterSetNameByResourceId,
+                  StringComparison.OrdinalIgnoreCase))
+                {
+                    var resourceId = new ResourceIdentifier(this.ResourceId);
+                    this.ResourceGroupName = resourceId.ResourceGroupName;
+                    this.Name = resourceId.ResourceName;
+                    var item = this.GetPeeringByResourceAndName();
+                    this.WriteObject(item);
+                }
+                else
+                {
+                    this.WriteObject(this.ListPeeringService(), true);
+                }
+            }
+            catch (InvalidOperationException mapException)
+            {
+                throw new InvalidOperationException(string.Format(Resources.Error_Mapping, mapException));
+            }
+            catch (ErrorResponseException ex)
+            {
+                var error = GetErrorCodeAndMessageFromArmOrErm(ex);
+                throw new ErrorResponseException(string.Format(Resources.Error_CloudError, error.Code, error.Message));
+            }
+        }
+
+        /// <summary>
+        ///  Lists Peering Services
+        /// </summary>
+        /// <returns>List of peering service resources</returns>
+        public List<PSPeeringService> ListPeeringService()
+        {
+            if(this.ShouldProcess(string.Format(Resources.ShouldProcessMessage, "a list of peering services for the subscription."))){
+                return this.PeeringServicesClient.ListBySubscription().Select(ToPeeringServicePS).ToList();
```

Comment:
@rileymckenna since the service supports listing by subscription, I would suggest the following:

Break the `ByResourceGroupAndName` parameter set into two:

- One with just an optional `-ResourceGroupName` parameter (to enable listing by subscription or resource group)
- One with mandatory `-ResourceGroupName` and `-Name` parameters (to enable getting a specific service)